test/testdata/rbi/range.rb

```diff
@@ -11,3 +11,7 @@
 (1.0...20.0)
 ('a'..'z')
 (:a...:z)
+
+# bsearch
+T.reveal_type((1..10).bsearch {|x| x == 1}) # error: Revealed type: `T.nilable(Integer)`
+T.reveal_type((1..10).bsearch {|x| x }) # error: Revealed type: `T.nilable(Integer)`
```

Comment:
Could you add a test that uses a non-`Integer` range as the receiver of `bsearch`?

Reply:
Done! Is that what you had in mind?

Reply:
The receiver is just the object that is the target of the method call. Thanks for adding these tests, they're exactly what I was thinking of!